### PR TITLE
Detection analytic fixes and enhancements

### PR DESCRIPTION
**Pull request by pemontto**
## Fixes

- Add missing operation `KeyDecrypt`
  - `Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml`

<s>- Missing severity for rules:
  - `Detections/AzureActivity/Creation_of_Expensive_Computes_in _Azure.yaml`
  - `Detections/MultipleDataSources/AnomalousIPUsageFollowedByTeamsAction.yaml`</s>

<s>- Missing attributes:
  - `Detections/MultipleDataSources/AnomalousIPUsageFollowedByTeamsAction.yaml`</s>

<s>- Maximum queryPeriod is 14d not 30d:
  - `Detections/MultipleDataSources/STRONTIUMOct292020IOCs.yaml`
  - `Detections/OfficeActivity/StrontiumCredHarvesting.yaml`</s>

<s>- TimeGenerated field is not available after summarise:
  - `Detections/SigninLogs/Sign-in Burst from Multiple Locations.yaml`</s>

## Enhancements

- Extract domain as host custom entity:
  - `Detections/CommonSecurityLog/MultiVendor-PossibleDGAContacts.yaml`

### CODE REVIEW

**Review by ashwin-patil — CHANGES_REQUESTED** [review]
Key vault detection was modified to address the noise hence `KeyDecrypt` was removed. 

Please revert back change to proceed.

File: Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ query: |
 
   let timeframe = 1d;
   let SensitiveOperationList = dynamic(
-  ["VaultDelete", "KeyDelete", "SecretDelete", "SecretPurge", "KeyPurge", "SecretBackup", "KeyBackup"]);
+  ["VaultDelete", "KeyDelete", "KeyDecrypt", "SecretDelete", "SecretPurge", "KeyPurge", "SecretBackup", "KeyBackup"]);
```
Comment (ashwin-patil) [inline Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml:23]: Please reference PR #849 where `KeyDecrypt` was removed due to being noisy as compared with other operations. 

We may have to monitor `KeyDecrypt` with the combination of IP Allowlist.

Comment (pemontto) [reply Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml]: For the moment I've changed the commit to remove `KeyDecrypt` from the description.

File: Detections/CommonSecurityLog/MultiVendor-PossibleDGAContacts.yaml [hunk 2]
```diff
@@ -111,4 +111,4 @@ query: |
     )
     on Name
     | project StartTime, EndTime, Name, DGADomain, SourceIP, DestinationIP, DataSource
-    | extend timestamp=StartTime, IPCustomEntity=SourceIP
\ No newline at end of file
+    | extend timestamp=StartTime, IPCustomEntity=SourceIP, HostCustomEntity=Name
```
Comment (shainw) [inline Detections/CommonSecurityLog/MultiVendor-PossibleDGAContacts.yaml:114]: Name in the instance is a Domain Name, not a Computer or Host Name.  We should not be pushing thru the Domain into this entity field.  We will soon have the ability to have a DNS DomainName field.  You can see the new changes coming here - https://github.com/Azure/Azure-Sentinel/pull/1319

**Review by shainw — CHANGES_REQUESTED** [review]

**Comment by shainw** [comment]
Closing and put change for keydecrypt remove in #1341.  Will submit a different PR for new entity mapping for PossibleDGAContacts detection.

**Pull request closed without merging**